File: app/src/lib/platform.ts
import { invoke } from '@tauri-apps/api/core';

export async function getPlatformInfo() {
  return invoke<{ os: string; arch: string; version: string }>('get_platform_info');
}

export async function isMobile() {
  return invoke<boolean>('is_mobile');
}

export function isTauri(): boolean {
  return '__TAURI__' in window;
}

File: app/src-tauri/src/commands/platform.rs
use serde::Serialize;

#[derive(Serialize)]
pub struct PlatformInfo {
    pub os: String,
    pub arch: String,
    pub version: String,
}

#[tauri::command]
pub fn get_platform_info() -> PlatformInfo {
    PlatformInfo {
        os: std::env::consts::OS.to_string(),
        arch: std::env::consts::ARCH.to_string(),
        version: env!("CARGO_PKG_VERSION").to_string(),
    }
}

#[tauri::command]
pub fn is_mobile() -> bool {
    cfg!(any(target_os = "android", target_os = "ios"))
}
